### PR TITLE
create V1 of article guardian lambda versions

### PR DESCRIPTION
**Pull request by PChol22**

### CODE REVIEW

File: blog-posts/guardian/lambda-versions/article.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+published: false
+title: 'AWS Lambda Versions : Time to clean up!'
```
Comment (ealain) [inline blog-posts/guardian/lambda-versions/article.md]: No space before colon.

File: blog-posts/guardian/lambda-versions/article.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+cover_image: https://raw.githubusercontent.com/pchol22/kumo-articles/master/blog-posts/guardian/lambda-versions/assets/cover_image.png
```
Comment (ealain) [inline blog-posts/guardian/lambda-versions/article.md]: Top!

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "dev.to",
   "repository": {
     "type": "git",
-    "url": "https://github.com/Kumo-by-Theodo/articles.git"
+    "url": "https://github.com/PChol22/kumo-articles.git"
```
Comment (ealain) [inline package.json:5]: You will be able to move this article to Kumo later, but don't forget to do it. You can ask Fred for the invitation link.

File: blog-posts/guardian/lambda-versions/article.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+# Let's keep track
+
+Lambda versioning is great and might come in handy, but do you really need to keep dozens of outdated copies of your functions' code ? Here are two reasons why you should get rid of them.
+
+## 1. Less is more
+
+It sure can help to have a backup version of your code to rollback to in case of an emergency, but the day it happens, you will feel a lot more conformable only having to chose between two or three documented, aliased versions than being overwhelmed by dozens of anonymous and forgotten ones.
+
+![Button meme](./assets/button_meme.png 'Button Meme')
```
Comment (ealain) [inline blog-posts/guardian/lambda-versions/article.md:22]: Nice!

File: blog-posts/guardian/lambda-versions/article.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## 2. Beware of AWS Lambda quotas
+
+AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you to past this ceiling in no time.
```
Comment (ealain) [inline blog-posts/guardian/lambda-versions/article.md]: Space: 75 GB

File: blog-posts/guardian/lambda-versions/article.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+It is possible to raise this soft limit by contacting the support, but it's not a sustainable workaround, and you don't want to be stuck unable to upload when there's an important fix to deploy.
+
+# Easily monitor your lambdas versioning with Guardian 🛡️
+
+[Guardian][guardian] now offers a **new rule** preventing your lambdas versions counter from going crazy.
+
+![Rule in CI](./assets/rule_CI.png 'Rule in CI')
```
Comment (ealain) [inline blog-posts/guardian/lambda-versions/article.md:34]: I think this screenshot does not put Guardian forward. Do you have the same with all green?

**Review by ealain — COMMENTED** [review]
Very nice to read!

File: blog-posts/guardian/lambda-versions/article.md (hunk 6 above)
Comment (PChol22) [reply blog-posts/guardian/lambda-versions/article.md:34]: that's true letme check

File: blog-posts/guardian/lambda-versions/article.md (hunk 1 above)
Comment (fredericbarthelet) [reply blog-posts/guardian/lambda-versions/article.md]: Ensure you have coherence between articles of the same serie.
For exemple, for Lift's articles, all titles ended with `Designing Lift`

What do you think of `AWS Lambda Versions : Time to clean up! - Guardian is watching over you` and also change @ealain article title to `AWS Lambda 101: Shave That Bundle Down - Guardian is watching over you`

File: blog-posts/guardian/lambda-versions/article.md (hunk 5 above)
Comment (fredericbarthelet) [inline blog-posts/guardian/lambda-versions/article.md]:
```suggestion
AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every Lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you over this threshold in no time.
```

Lambda always takes a capital `L`

File: blog-posts/guardian/lambda-versions/article.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+It sure can help to have a backup version of your code to rollback to in case of an emergency, but the day it happens, you will feel a lot more conformable only having to chose between two or three documented, aliased versions than being overwhelmed by dozens of anonymous and forgotten ones.
+
+![Button meme](./assets/button_meme.png 'Button Meme')
+
+## 2. Beware of AWS Lambda quotas
+
+AWS Lambda enforces a regional [75GB limit][quotas] for all your uploaded packages. This threshold might seem hard to reach and harmless, but it takes into account every version of every lambda you have deployed: dozens of versions per function multiplied by a large bundle size (see our [previous article][guardian-bundle-size-article]) will take you to past this ceiling in no time.
+
+It is possible to raise this soft limit by contacting the support, but it's not a sustainable workaround, and you don't want to be stuck unable to upload when there's an important fix to deploy.
```
Comment (fredericbarthelet) [inline blog-posts/guardian/lambda-versions/article.md]:
```suggestion
It is possible to raise this soft limit by contacting the support, but it's not a sustainable workaround, and you don't want to be stuck, unable to upload your new version, when there's an important fix to deploy.
```

File: blog-posts/guardian/lambda-versions/article.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+```
+npm install @kumo-by-theodo/guardian
+npx guardian -p <your_aws_profile> -c <your_stack_name>
+```
+
+Guardian is available on [NPM][npm-registry]. You will find instructions to use Guardian in your CI.
+
+# See also
+
+There also many other tools helping you manage your lambdas versioning. One of them is [Serverless Prune Plugin][serverless-prune-plugin], which integrates with the Serverless Framework and allows you to only keep a certain amount of recent versions for each one of your functions.
```
Comment (fredericbarthelet) [inline blog-posts/guardian/lambda-versions/article.md]:
```suggestion
There also many other tools helping you manage your lambdas versioning. One of them is [Serverless Prune Plugin][serverless-prune-plugin], which integrates with the Serverless Framework and allows you to only keep a certain amount of recent versions for each of your functions.
```

**Review by fredericbarthelet — COMMENTED** [review]
A few comments but otherwise great content :)! LGTM

File: blog-posts/guardian/lambda-versions/article.md (hunk 1 above)
Comment (ealain) [reply blog-posts/guardian/lambda-versions/article.md]: okpm

File: blog-posts/guardian/lambda-versions/article.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+_This article is part of a series on [Guardian][guardian], an open-source, highly configurable, automated best-practice audit tool for AWS serverless architectures._
+
+# Let's keep track
+
+Lambda versioning is great and might come in handy, but do you really need to keep dozens of outdated copies of your functions' code ? Here are two reasons why you should get rid of them.
+
+## 1. Less is more
+
+It sure can help to have a backup version of your code to rollback to in case of an emergency, but the day it happens, you will feel a lot more conformable only having to chose between two or three documented, aliased versions than being overwhelmed by dozens of anonymous and forgotten ones.
```
Comment (gozineb) [inline blog-posts/guardian/lambda-versions/article.md]: Tu voulais utiliser 'conformable' ou 'comfortable'?

Comment (PChol22) [reply blog-posts/guardian/lambda-versions/article.md]: ah oui en effet c'était comfortable

**Pull request merged**